pypesto/__init__.py

```diff
@@ -15,8 +15,8 @@
     CsvHistory,
     Hdf5History,
     OptimizerHistory,
-    Objective,
-    AmiciObjective)
+    AmiciObjective,
+    FunctionObjective)
```

Comment:
Looks really nice, I will have a closer look asap. One suggestion: In order not to "waste" the nice short name `Objective` on an abstract class, should we call `Objective` -> `ObjectiveBase` and `FunctionObjective` -> `Objective`? Then also the API would not change, i.e. people could continue using custom `Objective`s in their code, e.g. in the notebooks.

Reply:
Yeah, thought about that, increases the chances that someone checks `isinstance(obj, Objective)` and doesn't realise this excludes `AmiciObjective`, but thats probably easy and quick to spot.